### DISCUSSION TITLE
...Blazor com .NET 8 / ...prática / Editando Dados

### DISCUSSION THREAD

**Discussion by DrenokBR** [balta-io/forum#504] [Tira Dúvidas] [answered]
Fala Balta !

Acho que ainda não entendi tão bem o uso do "await", então perdoe-me se a reposta minha dúvida for muito óbvia, mas acho que com essa resposta devo entender melhor. Segue abaixo minha dúvida:

Por que você usou o "await" apenas na linha 50 e não na linha 49 e 51 conforme a imagem:

![await1](https://github.com/balta-io/forum/assets/36772075/4bc97e81-e525-4e48-8c1e-c2c54025c03b)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @DrenokBR, como vai?

Nem todos os métodos tem (ou precisam ter) uma versão assíncrona. O caso do SaveChanges, existe a versão SaveChangesAsync que utilizamos.

Uma vez que seu método é assíncrono, sempre escolha utilizar métodos `Async` dentro dele, é uma boa prática.

O `await` "aguarda" a execução de um método assíncrono.

**Reply by DrenokBR** [reply]
Aah ok. Entendi. Valeu Balta !

**Reply by DrenokBR** [reply]
Pelo que entendi só podemos trocar de página (linha 51) depois que o processo de salvar as alterações terminar (linha 50). É isso ?

E se for isso mesmo, o processo para salvar as alterações (linha 50) também não deveria esperar o processo de "update" (linha 49) terminar ? Ou seja, a linha 49 também não deveria receber um "await" ? Ou não se faz necessário porque o método .SaveChanges() já possui um mecanismo próprio de detecção de alterações no banco ?

**Reply by andrebaltieri** [reply]
@DrenokBR boa tarde, como vai? Conforme comento no curso de EF, sim, é necessário utilizar o SaveChanges para persistir os dados.
Dá uma revisada nas aulas do curso de EF que tem tudo explicado lá.

**Reply by DrenokBR** [reply]
Sim, eu lembro... eu fiz o curso recentemente. Mas vou dar mais uma olhada.
Valeu Balta, obrigadão !
